pingdom/check_types.go

```diff
@@ -318,6 +336,74 @@ func (ck *TCPCheck) Valid() error {
 	return nil
 }
 
+// PutParams returns a map of parameters for a DNSCheck that can be sent along
+// with an HTTP PUT request.
+func (ck *DNSCheck) PutParams() map[string]string {
+	m := map[string]string{
+		"name":             ck.Name,
+		"expectedip":       ck.ExpectedIp,
+		"host":             ck.Hostname,
+		"nameserver":       ck.NameServer,
+		"resolution":       strconv.Itoa(ck.Resolution),
+		"paused":           strconv.FormatBool(ck.Paused),
+		"notifyagainevery": strconv.Itoa(ck.NotifyAgainEvery),
+		"notifywhenbackup": strconv.FormatBool(ck.NotifyWhenBackup),
+		"integrationids":   intListToCDString(ck.IntegrationIds),
+		"probe_filters":    ck.ProbeFilters,
+		"tags":             ck.Tags,
+		"userids":          intListToCDString(ck.UserIds),
+		"teamids":          intListToCDString(ck.TeamIds),
+	}
+
+	if ck.SendNotificationWhenDown != 0 {
+		m["sendnotificationwhendown"] = strconv.Itoa(ck.SendNotificationWhenDown)
+	}
+
+	return m
+}
+
+// PostParams returns a map of parameters for a DNSCheck that can be sent along
+// with an HTTP POST request. Same as PUT.
+func (ck *DNSCheck) PostParams() map[string]string {
+	params := ck.PutParams()
+
+	for k, v := range params {
+		if v == "" {
+			delete(params, k)
+		}
+	}
+
+	params["type"] = "dns"
+	return params
+}
+
+// Valid determines whether the DNSCheck contains valid fields.  This can be
+// used to guard against sending illegal values to the Pingdom API.
+func (ck *DNSCheck) Valid() error {
+	if ck.Name == "" {
+		return fmt.Errorf("invalid value for `Name`, must contain non-empty string")
+	}
+
+	if ck.ExpectedIp == "" {
+		return fmt.Errorf("invalud value for `ExpectedIp`, must containt non-empty string")
+	}
+
+	if ck.Hostname == "" {
+		return fmt.Errorf("invalid value for `Hostname`, must contain non-empty string")
+	}
+
+	if ck.NameServer == "" {
+		return fmt.Errorf("invalid value for `NameServer`, must contain non-empty string")
+	}
+
+	if ck.Resolution != 1 && ck.Resolution != 5 && ck.Resolution != 15 &&
+		ck.Resolution != 30 && ck.Resolution != 60 {
+		return fmt.Errorf("invalid value %v for `Resolution`, allowed values are [1,5,15,30,60]", ck.Resolution)
+	}
+
+	return nil
+}
+
 func intListToCDString(integers []int) string {
```

Comment:
Not sure if you want to add comments here `//` since you do it for other `functions`.

However if you feel it is implicit enough, don't mind me.

**Some reading (regarding it looks good in your code):**
Was doing some reading on `golang` commenting, quite interesting you simply just need to do `//`  above a `function` and it will be visible in the docs.

https://godoc.org/github.com/natefinch/godocgo